src/main/java/org/opensearch/knn/index/codec/KNN990Codec/KNN990HnswBinaryVectorsFormat.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.KNN990Codec;
+
+import org.apache.lucene.codecs.KnnVectorsFormat;
+import org.apache.lucene.codecs.KnnVectorsReader;
+import org.apache.lucene.codecs.KnnVectorsWriter;
+import org.apache.lucene.codecs.hnsw.FlatVectorsFormat;
+import org.apache.lucene.codecs.lucene99.Lucene99FlatVectorsFormat;
+import org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsReader;
+import org.apache.lucene.codecs.lucene99.Lucene99HnswVectorsWriter;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SegmentWriteState;
+import org.apache.lucene.search.TaskExecutor;
+
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+
+public class KNN990HnswBinaryVectorsFormat extends KnnVectorsFormat {
```

Comment:
I also wasn't sure on this. The `KNN9120Codec` codec maps to the `KNN990PerFieldKnnVectorsFormat` class so I felt `KNN990HnswBinaryVectorsFormat` was more in line with what was present.

Reply:
@jed326, `KNN990PerFieldKnnVectorsFormat` is for older version of lucene. I think since you are making changes in the format code too you should create a `KNN9120PerFieldKnnVectorsFormat` class and then `KNN9120HnswBinaryVectorsFormat` for consistency. @jmazanec15 please comment if you think otherwise.